addons/hr_expense/tests/test_expenses.py

```diff
@@ -1404,3 +1404,32 @@ def test_multiple_payments(self):
             {'balance':   199.42, 'amount_currency':   130.43, 'currency_id': foreign_currency.id},
             {'balance': -1528.90, 'amount_currency': -1000.00, 'currency_id': foreign_currency.id},
         ])
+
+    def test_currency_rate_override(self):
+        """ Checks that the currency rate is recomputed properly when the total in company currency is set to a new value """
+        expense = self.env['hr.expense'].create({
+            'name': 'Company expense',
+            'date': '2022-11-16',
+            'payment_mode': 'company_account',
+            'total_amount': 1000.00,
+            'employee_id': self.expense_employee.id,
+            'product_id': self.product_c.id,
+                'currency_id': self.currency_data['currency'].id,  # rate is 1:2
+        })
+        self.assertEqual(0.5, expense.currency_rate)
+        expense.write({'total_amount_company': 1000.00})
+        self.assertEqual(1.0, expense.currency_rate)
+
+        expense_sheet = self.env['hr.expense.sheet'].create({
+            'name': 'Expense for Company',
+            'employee_id': self.expense_employee.id,
+            'expense_line_ids': expense,
+        })
```

Comment:
Could we test multiple expenses with different exchange rates? also verify that the company exchange rate differs

Reply:
> Could we test multiple expenses with different exchange rates? 
-> Done

> also verify that the company exchange rate differs
-> Why ?